client/asset/dcr/dcr.go

```diff
@@ -4690,11 +4690,12 @@ func (dcr *ExchangeWallet) handleTipChange(ctx context.Context, newTipHash *chai
 	}
 
 	dcr.tipMtx.Lock()
-	defer dcr.tipMtx.Unlock()
-
 	prevTip := dcr.currentTip
 	dcr.currentTip = &block{newTipHeight, newTipHash}
+	dcr.tipMtx.Unlock()
```

Comment:
Doesn't seem like there is a need to hold this mutex for the rest of the **func** here ?

Reply:
Maybe just to avoid running `handleTipChange` concurrently for example, such that 2nd update will run first, resulting in old tip rewriting the latest one. Reverted.